File: src/org/cappuccino_project/ide/intellij/plugin/indices/ObjJImplementationDeclarationsIndex.java
package org.cappuccino_project.ide.intellij.plugin.indices;

import com.intellij.psi.stubs.StubIndexKey;
import org.cappuccino_project.ide.intellij.plugin.psi.ObjJImplementationDeclaration;
import org.jetbrains.annotations.NotNull;

public class ObjJImplementationDeclarationsIndex extends ObjJStringStubIndexBase<ObjJImplementationDeclaration> {

    private static final ObjJImplementationDeclarationsIndex INSTANCE = new ObjJImplementationDeclarationsIndex();
    private static final StubIndexKey<String, ObjJImplementationDeclaration> KEY = IndexKeyUtil.createIndexKey(ObjJImplementationDeclarationsIndex.class);
    private static final int VERSION = 0;

    private ObjJImplementationDeclarationsIndex(){}

    public static ObjJImplementationDeclarationsIndex getInstance() {
        return INSTANCE;
    }

    @Override
    public int getVersion() {
        return super.getVersion() + VERSION;
    }

    @NotNull
    @Override
    protected Class<ObjJImplementationDeclaration> getIndexedElementClass() {
        return ObjJImplementationDeclaration.class;
    }

    @NotNull
    @Override
    public StubIndexKey<String, ObjJImplementationDeclaration> getKey() {
        return KEY;
    }

}

File: src/cappuccino/ide/intellij/plugin/indices/ObjJImplementationDeclarationsIndex.kt
package cappuccino.ide.intellij.plugin.indices

import cappuccino.ide.intellij.plugin.psi.ObjJImplementationDeclaration
import com.intellij.psi.stubs.StubIndexKey

class ObjJImplementationDeclarationsIndex private constructor() : ObjJStringStubIndexBase<ObjJImplementationDeclaration>() {

    override val indexedElementClass: Class<ObjJImplementationDeclaration>
        get() = ObjJImplementationDeclaration::class.java

    override fun getVersion(): Int {
        return super.getVersion() + VERSION
    }

    override fun getKey(): StubIndexKey<String, ObjJImplementationDeclaration> {
        return KEY
    }

    companion object {

        val instance = ObjJImplementationDeclarationsIndex()
        private val KEY = IndexKeyUtil.createIndexKey(ObjJImplementationDeclarationsIndex::class.java)
        private const val VERSION = 3
    }

}
